x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/EqlPlugin.java

```diff
@@ -60,6 +87,11 @@ boolean isSnapshot() {
         return Build.CURRENT.isSnapshot();
     }
 
+    // TODO: this needs to be used by all plugin methods - including getActions and createComponents
```

Comment:
probably address this TODO before merging to master?

Reply:
I don't think there's a way to do that currently - the solution is to have a constructor that accepts settings however we provide our settings after instantiation ...